fluss-server/src/main/java/com/alibaba/fluss/server/log/LogTablet.java

```diff
@@ -1115,7 +1115,8 @@ private static void updateWriterAppendInfo(
         // update writers.
         WriterAppendInfo appendInfo =
                 writers.computeIfAbsent(writerId, id -> writerStateManager.prepareUpdate(writerId));
-        appendInfo.append(batch);
+        appendInfo.append(
+                batch, writerStateManager.isBatchExpired(System.currentTimeMillis(), batch));
```

Comment:
Can we move the calculation of `writerStateManager.isBatchExpired` to `loadWritersFromRecords`? This can avoid too many calls of `System.currentTimeMillis()` for each record batch and only checks `isBatchExpired` for restoring state from records.

Reply:
It seems we need to check both for restoring state from records and appending log as follower.